### ISSUE TITLE
[BART on CNN/DM] issue on load checkpoint

### ISSUE THREAD

**Issue by Liatwilight** [pytorch/fairseq#1476]
When I try to finetune the model following the README (https://github.com/pytorch/fairseq/blob/master/examples/bart/README.cnn.md), I met some problems:
```plaintext
Traceback (most recent call last):
  File "train.py", line 337, in <module>
    cli_main()
  File "train.py", line 329, in cli_main
    nprocs=args.distributed_world_size,
  File "/home/guimin.gm/miniconda3/envs/pt/lib/python3.6/site-packages/torch/multiprocessing/spawn.py", line 171, in spawn
    while not spawn_context.join():
  File "/home/guimin.gm/miniconda3/envs/pt/lib/python3.6/site-packages/torch/multiprocessing/spawn.py", line 118, in join
    raise Exception(msg)
Exception:

-- Process 3 terminated with the following error:
Traceback (most recent call last):
  File "/home/guimin.gm/miniconda3/envs/pt/lib/python3.6/site-packages/torch/multiprocessing/spawn.py", line 19, in _wrap
    fn(i, *args)
  File "/gruntdata/guimin.gm/bart/fairseq-master/train.py", line 296, in distributed_main
    main(args, init_distributed=True)
  File "/gruntdata/guimin.gm/bart/fairseq-master/train.py", line 70, in main
    extra_state, epoch_itr = checkpoint_utils.load_checkpoint(args, trainer)
  File "/gruntdata/guimin.gm/bart/fairseq-master/fairseq/checkpoint_utils.py", line 115, in load_checkpoint
    reset_meters=args.reset_meters,
  File "/gruntdata/guimin.gm/bart/fairseq-master/fairseq/trainer.py", line 186, in load_checkpoint
    state = checkpoint_utils.load_checkpoint_to_cpu(filename)
  File "/gruntdata/guimin.gm/bart/fairseq-master/fairseq/checkpoint_utils.py", line 146, in load_checkpoint_to_cpu
    with PathManager.open(path, "rb") as f:
  File "/gruntdata/guimin.gm/bart/fairseq-master/fairseq/file_io.py", line 51, in open
    newline=newline,
TypeError: Required argument 'file' (pos 1) not found
```
I print the value of the parameter "path" here, it equals to "/gruntdata/guimin.gm/bart/fairseq-master/bart-pt/bart.large.cnn/model.pt", where has saved the official pretrained bart model.

I am checking this on
```
Python 3.6.8
pytorch '1.2.0+cu92'
```

Would you share any thoughts on the matter? It would help me a lot.
Thank you very much!

**Comment by lematt1991** [comment]
Can you include the full command you used to launch the job

**Comment by Liatwilight** [comment]
I finetune the model using the command:
```shell
TOTAL_NUM_UPDATES=20000
WARMUP_UPDATES=500
LR=3e-05
MAX_TOKENS=1024
UPDATE_FREQ=4
BART_PATH=/gruntdata/guimin.gm/bart/fairseq-master/bart-pt/bart.large.cnn/model.pt

CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 python train.py /home/guimin.gm/cnndm/cnndm-data/finished_files/cnn_dm-bin \
    --restore-file $BART_PATH \
    --max-tokens $MAX_TOKENS \
    --task translation \
    --source-lang source --target-lang target \
    --truncate-source \
    --layernorm-embedding \
    --share-all-embeddings \
    --share-decoder-input-output-embed \
    --reset-optimizer --reset-dataloader --reset-meters \
    --required-batch-size-multiple 1 \
    --fp16 --arch bart_large \
    --criterion label_smoothed_cross_entropy \
    --label-smoothing 0.1 \
    --dropout 0.1 --attention-dropout 0.1 \
    --weight-decay 0.01 --optimizer adam --adam-betas "(0.9, 0.999)" --adam-eps 1e-08 \
    --clip-norm 0.1 \
    --lr-scheduler polynomial_decay --lr $LR --total-num-update $TOTAL_NUM_UPDATES --warmup-updates $WARMUP_UPDATES \
    --update-freq $UPDATE_FREQ \
    --skip-invalid-size-inputs-valid-test \
    --find-unused-parameters;
```
I found that something went wrong with these two lines of code:

[https://github.com/pytorch/fairseq/blob/master/fairseq/checkpoint_utils.py#L145](https://github.com/pytorch/fairseq/blob/master/fairseq/checkpoint_utils.py#L145)
[https://github.com/pytorch/fairseq/blob/master/fairseq/checkpoint_utils.py#L271](https://github.com/pytorch/fairseq/blob/master/fairseq/checkpoint_utils.py#L271)

I modified this two lines:
```python
with open(path,"rb") as f:
with open(filename, "wb") as f:
```

**Then the problem disappeared. But I don’t know why this kind of problem occurs, and whether there are hidden dangers if I make the above changes.**

Now I have successfully reproduced the performance on CNN/DM dataset, and I am trying to finetune the model.

**Comment by myleott** [comment]
This was fixed in af3050285d646125e7c76a245188b23a0b043185 last week.
